warehouse/db.go

```diff
@@ -0,0 +1,50 @@
+package warehouse
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	warehouseutils "github.com/rudderlabs/rudder-server/warehouse/utils"
+)
+
+// DB encapsulate interactions of warehouse operations
+// with the database.
+type DB struct {
+	handle *sql.DB
+}
+
+func NewWarehouseDB(handle *sql.DB) *DB {
```

Comment:
Can we expose an interface here instead of calling it from a struct? wdyt
Later we can easily mock the interface.

Reply:
Also, Can we move this into different package? Because since we will be moving all queries and these files becomes big?

Reply:
Good point about the interface. The reason I didn't add the interface was because it's a starting point and for interface addition, we might need to break the queries further.

Similar point about a separate module. Given it grows beyond a certain point and many supporting structs are needed, it might be good to get it into a separate module.

Reply:
Thanks @abhimanyubabbar for sharing your thoughs.